client_credentials_auth.go

```diff
@@ -0,0 +1,104 @@
+package sdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type ClientCredentialsTokenSource struct {
+	clientID     string
+	clientSecret string
+	tokenURL     string
+	scope        string
+	grantType    string
+	token        *ClientCredentialsToken
```

Comment:
Should there be a guard on the token for when multiple go routines try to access it when calling the `Token` method?

Like we have in the IAM auth implementation: https://github.com/openfaas/go-sdk/blob/73f84ff5d81d87ba6d1ea5c8672ebb72600966d3/auth.go#L39C2-L39C2